src/xo-app/xoa-updates/index.js

```diff
@@ -0,0 +1,279 @@
+  _handleProxyUserChange = event => this.setState({proxyUser: event.target.value || ''})
+
+  _handleConfigReset = () => {
+    const { configuration } = this.props
+    const { proxyPassword } = this.refs
+    proxyPassword.value = ''
+    this.setState(configuration)
+  }
+
+  _register = async () => {
+    const {
+      email,
+      password
+    } = this.refs
+
+    const { registration } = this.props
+    const alreadyRegistered = (registration.state === 'registered')
+    if (alreadyRegistered) {
+      try {
+        await confirm({
+          title: 'Replace registration ?',
+          body: <p>
+          Your XO appliance is already registered to {registration.email}, do you want to forget and replace this registration ?
+          </p>})
+      } catch (error) {
+        return
+      }
+      return xoaUpdater.register(email.value, password.value, alreadyRegistered)
+      .then(() => { email.value = password.value = '' })
+    }
+  }
+
+  _configure = async () => {
+    const {
+      proxyHost,
+      proxyPort,
+      proxyUser
+    } = this.state
+    const { proxyPassword } = this.refs
+    return xoaUpdater.configure({
+      proxyHost,
+      proxyPort,
+      proxyUser,
+      proxyPassword: proxyPassword.value
+    })
+    .then(config => {
+      this.setState({
+        proxyHost: undefined,
+        proxyPort: undefined,
+        proxyUser: undefined
+      })
+      proxyPassword.value = ''
+    })
+  }
+
+  _startTrial = async () => {
+    try {
+      await confirm({
+        title: 'Ready for trial ?',
+        body: <p>During the trial period, XOA need to have a working internet connection. This limitation does not apply for our paid plans!</p>
+      })
+      return xoaUpdater.requestTrial()
+        .then(() => xoaUpdater.update())
+        .catch(err => error('Request Trial', err.message || String(err)))
+    } catch (_) {
+      return
+    }
+  }
+
+  render () {
+    const textClasses = {
+      info: 'text-info',
+      success: 'text-success',
+      warning: 'text-warning',
+      error: 'text-danger'
+    }
+
+    const {
+      log,
+      registration,
+      state,
+      trial
+    } = this.props
+    let { configuration } = this.props // Configuration from the store
+
+    configuration = assign({}, configuration)
+    const {
+      proxyHost,
+      proxyPort,
+      proxyUser
+    } = this.state // Edited non-saved configuration values override in view
+    let configEdited = false
+    proxyHost !== undefined && (configuration.proxyHost = proxyHost) && (configEdited = true)
+    proxyPort !== undefined && (configuration.proxyPort = proxyPort) && (configEdited = true)
+    proxyUser !== undefined && (configuration.proxyUser = proxyUser) && (configEdited = true)
+
+    return <div className='container-fluid'>
+      <p>
+        <strong>{states[state]}</strong>
+        {' '}
+        <ActionButton
+          btnStyle='info'
+          handler={() => xoaUpdater.update()}
```

Comment:
Avoid creating values (arrays, functions, objects) in render, I can add an `handlerContext` if you want.